src/dnne-analyzers/AttributesGenerator.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace DNNE;
+
+/// <summary>
+/// A generator that generates all the necessary DNNE attributes.
+/// </summary>
+[Generator(LanguageNames.CSharp)]
+public sealed class AttributesGenerator : IIncrementalGenerator
+{
+    /// <inheritdoc/>
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(static context =>
+        {
+            context.AddSource("DnneAttributes.g.cs", """
+                // <auto-generated/>
+                #pragma warning disable
+
+                namespace DNNE
+                {
+                    /// <summary>
+                    /// Defines a C export. Can be used when updating to use <c>UnmanagedCallersOnlyAttribute</c> would take more time.
+                    /// </summary>
+                    internal sealed class ExportAttribute : global::System.Attribute
```

Comment:
```suggestion
                    [System.AttributeUsage(System.AttributeTargets.Method, Inherited=false)]
                    internal sealed class ExportAttribute : global::System.Attribute
```

Reply:
Updated with `[AttributeUsage]` for all 3 generated types 👍